com.unity.formats.fbx/Editor/FbxExporter.cs

```diff
@@ -1953,18 +1966,39 @@ private FbxConstraint GetFbxConstraint(FbxNode constrainedNode, System.Type uniC
                 
             foreach (var constraint in constraints)
             {
-                if (uniConstraintType != constraint.Value)
+                if (uniConstraintType == constraint.Value)
                 {
-                    continue;
+                    return constraint.Key;
                 }
+            }
+            return null;
+        }
 
-                return constraint.Key;
+        /// <summary>
+        /// Get the FbxBlendshape with the given name associated with the FbxNode.
+        /// </summary>
+        /// <param name="blendshapeNode"></param>
+        /// <param name="uniPropertyName"></param>
+        /// <returns></returns>
+        private FbxBlendShapeChannel GetFbxBlendShape(FbxNode blendshapeNode, string uniPropertyName)
+        {
+            Dictionary<FbxBlendShapeChannel, string> blendshapeChannels;
+            if (!MapUnityObjectToBlendShapes.TryGetValue(blendshapeNode, out blendshapeChannels))
+            {
+                return null;
             }
 
+            foreach (var channel in blendshapeChannels)
+            {
+                if (uniPropertyName == ("blendShape." + channel.Value))
```

Comment:
Parse uniPropertyName and remove the `blendShape.` so the comparison is just an == and doesn't require creating a string on every channel.

This can be `firstOrDefault` as well.